tests/test_connector_valetudo_events.py

```
@@ -0,0 +1,436 @@
+
+    # Collect coroutines scheduled via hass.async_create_task so we can await them.
+    scheduled_coros: List[Any] = []
+    connector.connector_data.hass.async_create_task = lambda coro: scheduled_coros.append(coro)
+
+    captured_callbacks: List[Any] = []
+
+    def _capture(_hass, cb):
+        captured_callbacks.append(cb)
+        return MagicMock()
+
+    # Step 1–2: Valetudo sends an unprocessed error.
```

Ruff (0.15.11)

[warning] Comment contains ambiguous `–` (EN DASH). Did you mean `-` (HYPHEN-MINUS)?

(RUF003)